web/src/ApplicationEditPage.js

```diff
@@ -119,14 +119,17 @@ class ApplicationEditPage extends React.Component {
   getApplication() {
     ApplicationBackend.getApplication("admin", this.state.applicationName)
       .then((application) => {
-        if (application.grantTypes === null || application.grantTypes === undefined || application.grantTypes.length === 0) {
-          application.grantTypes = ["authorization_code"];
+        if (application === undefined || application === null) {
+          this.props.history.push("/404");
+        } else {
```

Comment:
Just add the following code (with a trailing new line), use `return` in if, don't add `else` and change other code

```js
if (application === null) {
  this.props.history.push("/404");
  return;
}
```